lib/reservation_monitor.py

```diff
@@ -194,42 +197,63 @@ def _check(self) -> bool:
-            )
-            self.notification_handler.timeout_during_retrieval("account")
-            return [], True
-        except LoginError as err:
-            if err.status_code == TOO_MANY_REQUESTS_CODE:
-                # Don't exit when a Too Many Requests error happens. Instead, just skip the
-                # retrieval until the next time.
+        for attempt in range(max_retries + 1):
+            webdriver = WebDriver(self.checkin_scheduler)
+
+            try:
+                reservations = webdriver.get_reservations(self)
                 logger.debug(
-                    "Encountered a Too Many Requests error while logging in. Skipping reservation "
-                    "retrieval"
+                    "Successfully retrieved %d reservations after %d attempts",
+                    len(reservations),
+                    attempt + 1,
                 )
-                self.notification_handler.too_many_requests_during_login()
-                return [], True
-
-            logger.debug("Error logging in. %s. Exiting", err)
-            self.notification_handler.failed_login(err)
-            sys.exit(1)
-
-        logger.debug("Successfully retrieved %d reservations", len(reservations))
-        return reservations, False
+                return reservations, False
+
+            except DriverTimeoutError:
+                if attempt < max_retries:
+                    logger.debug("Timeout while retrieving reservations during login. Retrying")
+                    logger.debug("Waiting for %d seconds before retrying", RETRY_WAIT_SECONDS)
+                    time.sleep(RETRY_WAIT_SECONDS)
+                else:
+                    logger.debug(
+                        "Timeout persisted after %d retries. Skipping reservation retrieval",
+                        max_retries,
+                    )
+                    self.notification_handler.timeout_during_retrieval("account")
+
+            except LoginError as err:
+                if err.status_code in [TOO_MANY_REQUESTS_CODE, INTERNAL_SERVER_ERROR_CODE]:
+                    if attempt < max_retries:
+                        logger.debug(
+                            "Encountered an error (status: %d) while logging in. Retrying",
+                            err.status_code,
+                        )
+                        logger.debug("Waiting for %d seconds before retrying", RETRY_WAIT_SECONDS)
+                        time.sleep(RETRY_WAIT_SECONDS)
+                    else:
+                        logger.debug(
+                            "Error (status: %d) persists. Skipping reservation retrieval",
+                            err.status_code,
+                        )
+                        self.notification_handler.too_many_requests_during_login()
+                else:
+                    logger.debug("Error logging in. %s. Exiting", err)
+                    self.notification_handler.failed_login(err)
+                    time.sleep(1)
```

Comment:
Is this sleep call needed? The notification has still been sent successfully if I remove the call (I believe it is all done synchronously). It's a very small nitpick though, so I can just remove it later if we decide that it isn't needed.

Reply:
We can probably remove it, I don't remember why it's there )